niri-ipc/src/lib.rs

```diff
@@ -475,6 +484,20 @@ pub enum OutputConfigChanged {
     OutputWasMissing,
 }
 
+impl FromStr for WorkspaceReferenceArg {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let reference = if let Ok(index) = s.parse::<u8>() {
```

Comment:
I had an idea to parse an `i32` then error out if it cannot be converted to `u8`. For future extensibility (maybe we'll want negative indices in the future, who knows) and to ease out mistakes where people set > 255 by accident.

Reply:
I'd keep it as u8 for now, so it is aligned with the current implementation. However, for future extensibility, we could introduce a type alias, and use it wherever a workspace index is to be used. That way, if you decide to use an `i32` in the future, it will have to be changed in one place only.

What do you think?

Reply:
My reasoning for this is to prevent people from using these numbers as workspace names, so that if we do decide to use them in the future, we can do so without breaking existing scripts.

Reply:
Ah, I see! Yeah, that makes sense. I'll update it in a bit.